### ISSUE TITLE
BUG: import numpy causes python to crash without any error

### ISSUE THREAD

**Issue by GurpreetSarangal** [numpy/numpy#23074]
### Describe the issue:

Actually I am starting to learn OpenCV, when i imported `cv2`, it caused python to crash, then I investigated the cv2 package and found that cv2 internally imports numpy, which was causing the crash. So on checking the `__init__.py` file of `numpy`, I noticed that the statement on line 189 in, `numpy > __init__.py`
```python
core.getlimits._register_known_types()
```
was causing this strange thing. On exploration of source code, I found out that `core > getlimits.py > class MachArLike > __init__() ` was causing it
```python
self.resolution = self._float_to_float(
     self._float_conv(10) ** (-self.precision))
```
After some hit and try, I have figured out there is just a minor change required in this statement
```python
self.resolution = self._float_to_float(
     self._float_conv(10)[0] ** (-self.precision))
```
Please assign this bug to me, I would love to contribute to numpy community.

### Reproduce the code example:

```python
a = 12
b = 2131
print(a+b)

try:
    import numpy
    print("numpy imported")
except:
    print("didn't imported")
```

### Error message:

```shell
no error message, program just abruptly ends like this

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python -u "c:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice\simple.py"
2143

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>
```

### Runtime information:

>>> import sys, numpy; print(numpy.__version__); print(sys.version)
1.24.1
3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:34:34) [MSC v.1928 32 bit (Intel)]
>>> print(numpy.show_runtime())
WARNING: `threadpoolctl` not found in system! Install it by `pip install threadpoolctl`. Once installed, try `np.show_runtime` again for more detailed build information
[{'simd_extensions': {'baseline': ['SSE', 'SSE2'],
                      'found': ['SSE3', 'SSSE3', 'SSE41', 'POPCNT', 'SSE42'],
                      'not_found': ['AVX',
                                    'F16C',
                                    'FMA3',
                                    'AVX2',
                                    'AVX512F',
                                    'AVX512CD',
                                    'AVX512_SKX',
                                    'AVX512_CLX',
                                    'AVX512_CNL',
                                    'AVX512_ICL']}}]
None

**Comment by GurpreetSarangal** [comment]
is `numpy` different for 32-bit and 64-bit computers?

**Comment by GurpreetSarangal** [comment]
I just installed `opencv-python` by this command
`pip install opencv-python`
numpy was automatically installed as is required by `cv2`

**Comment by GurpreetSarangal** [comment]
btw, I am using a 32-bit computer

**Comment by mattip** [comment]
This works
```
import sys, numpy; print(numpy.version)
```

So it is some interaction betweeen opencv-python and NumPy. The code you point to `self._float_conv(10)` calls `array([10], self.ftype)`, and so somehow on your machine `array([value], self.ftype) ** (-self.precision))` is segfaulting. When you call `self._float_conv(10)[0]` that is returning the python integer `10` (rather than a NumPy `array`) so `10 ** (-self.precision)` succeeds.

Could you print out `pip list`?

**Comment by GurpreetSarangal** [comment]
Output of `pip list`
```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>pip list
Package       Version
------------- --------
numpy         1.24.1
opencv-python 4.7.0.68
pip           22.3.1
setuptools    65.6.3
wheel         0.38.4
```

**Comment by GurpreetSarangal** [comment]
to check this `import sys, numpy; print(numpy.version)`, I reverted my changes but still the shell itself ended abruptly. 
output in shell mode
```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python
Python 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:34:34) [MSC v.1928 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys, numpy; print(numpy.version)

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>
```

**Comment by GurpreetSarangal** [comment]
however, with my changes, `import sys, numpy; print(numpy.version)` this outputs
```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python
Python 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:34:34) [MSC v.1928 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys, numpy; print(numpy.version)
<module 'numpy.version' from 'C:\\Users\\ashok\\Documents\\_StudyMaterial\\Code\\Face Recognition - Open CV practice\\venv\\lib\\site-packages\\numpy\\version.py'>
>>>
```

**Comment by GurpreetSarangal** [comment]
is the statement `self.resolution = self._float_to_float(self._float_conv(10) ** (-self.precision))` working fine on your system?

**Comment by mattip** [comment]
It works fine for me, CI, and lot of other people, otherwise we would have gotten more complaints. Is this inside a virtual machine or some other emulation? Perhaps it is similar to #22979

**Comment by GurpreetSarangal** [comment]
No, it is on my system, Windows 7 32-bit. I am not using any VM or emulator. Actually, it is quite similar to #22979.  
Would you need some more information about my system? like they provided in this issue. If so, please also give me commands to get that as I am just a beginner here.

**Comment by mattip** [comment]
NumPy has an array class, it is very different from `[10]` which is a list containing an integer. When you do `np.array([10]) ** 3` you call into a [ufunc](https://numpy.org/devdocs/glossary.html#term-ufunc) that is set up to repeatedly apply exponentiation on values in a fast manner. One of the things that "fast manner" does is utilize [hardware acceleration](https://numpy.org/devdocs/reference/simd/index.html) called SIMD. Your machine showed it can use 'SSE3', 'SSSE3', 'SSE41', 'POPCNT', 'SSE42' intrinsics to speed up operations. Perhaps in both the VM issue in #22979 and here, the CPU is claiming it properly supports SIMD intrinsics, but in practice the support is faulty.

Unfortunately, NumPy barely supports 32-bit executables running on 64-bit windows. I am afraid it will be very difficult to figure out what is going wrong with a 32-bit version of Windows7. What hardware are you running on? You may want to try to use a version of NumPy without the SIMD optimizations, I think 1.19 was the last version without them. If `pip install "numpy<1.20" gets you a working version of NumPy, then maybe you can find a version of opencv-python that can work with that.

**Comment by seberg** [comment]
Curious, what happens if you just call:
```
10.0**-500
```
in Python, and if that works:
```
np.float64(10.0)**-500
```

I have seen issues at a similar point, but they were embededded Python shells setting up floating point exceptions in a bad way.

**Comment by GurpreetSarangal** [comment]
hey, I just noticed something, when `np.array([10.0], np.float16) ** 3` or `np.array([10.0], np.float32) ** 3` is called, it crashed python. But when I call `np.array([10.0], np.float64) ** 3` it outputs `array([1000.])`, similarly when `array([10]) ** 3` is called it also outputs `array([1000.])`  
and when I called `np.array([10], np.float16)[0] ** 3` it prints `1000.0`
```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python
Python 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:34:34) [MSC v.1928 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> np.array([10]) ** 3
array([1000], dtype=int32)
>>> np.array([10.0]) ** 3
array([1000.])
>>> np.array([10], np.float16) ** 3

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python
Python 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:34:34) [MSC v.1928 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> np.array([10]) ** 3
array([1000], dtype=int32)
>>> np.array([10.0]) ** 3
array([1000.])
>>> np.array([10], np.float32) ** 3

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python
Python 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:34:34) [MSC v.1928 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> np.array([10], np.float64) ** 3
array([1000.])
>>> np.array([10], np.float16)[0] ** 3
1000.0
>>>
```

**Comment by GurpreetSarangal** [comment]
hey @mattip you asked for hardware details, can you explain exactly what details are you asking about?

**Comment by GurpreetSarangal** [comment]
I think this is what you were asking for:
```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python
Python 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:34:34) [MSC v.1928 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> np.show_runtime()
[{'simd_extensions': {'baseline': ['SSE', 'SSE2'],
                      'found': ['SSE3', 'SSSE3', 'SSE41', 'POPCNT', 'SSE42'],
                      'not_found': ['AVX',
                                    'F16C',
                                    'FMA3',
                                    'AVX2',
                                    'AVX512F',
                                    'AVX512CD',
                                    'AVX512_SKX',
                                    'AVX512_CLX',
                                    'AVX512_CNL',
                                    'AVX512_ICL']}},
 {'architecture': 'Nehalem',
  'filepath': 'C:\\Users\\ashok\\Documents\\_StudyMaterial\\Code\\Face '
              'Recognition - Open CV '
              'practice\\venv\\Lib\\site-packages\\numpy\\.libs\\libopenblas_v0.3.21-gcc_8_3_0.dll',
  'internal_api': 'openblas',
  'num_threads': 4,
  'prefix': 'libopenblas',
  'threading_layer': 'pthreads',
  'user_api': 'blas',
  'version': '0.3.21'}]
>>>
```

**Comment by GurpreetSarangal** [comment]
I forgot to mention, I have also changed this import statement in `numpy > core > getlimits.py` 
```python
# import statement before I changed
from .umath import log10, exp2, nextafter, isnan

# what I changed it to
from .umath import exp2, nextafter, isnan
from math import log10 
.
.
.
self.precision = int(-log10(self.eps))
```
before I changed the following statement, `log10()` was crashing the system, so I changed the import statement.  
```
self.resolution = self._float_to_float(
     self._float_conv(10) ** (-self.precision))
```

After changing import, `self._float_conv(10) ** (-self.precision)` started causing system to crash.

**Comment by mattip** [comment]
could you try setting these before running python?
```
C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET OPENBLAS_CORETYPE=Haswell
C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET OPENBLAS_NUM_THREADS=1
C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET NPY_DISABLE_CPU_FEATURES="SSE3,SSSE3,SSE41,POPCNT,SSE42
```

Then run the original code with no modifications
```
C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python -c "import numpy; numpy.show_runtime()"
```

**Comment by GurpreetSarangal** [comment]
the output of those commands, 
```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET OPENBLAS_CORETYPE=Haswell

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET OPENBLAS_NUM_THREADS=1

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET NPY_DISABLE_CPU_FEATURES="SSE3,SSSE3,SSE41,POPCNT,SSE42

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python -c "import numpy; numpy.show_runtime()"
<frozen importlib._bootstrap>:219: RuntimeWarning: During parsing environment variable 'NPY_DISABLE_CPU_FEATURES':
You cannot disable CPU features ("SSE3), since they are not part of the dispatched optimizations
(SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512CD AVX512_SKX AVX512_CLX AVX512_CNL AVX512_ICL).

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>
```
Before running these, as you said, I reverted my changes.   
Also, could you explain those commands? or just give me some resources where I can learn about those.

**Comment by GurpreetSarangal** [comment]
I think this is what you meant by the last command you gave
```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET NPY_DISABLE_CPU_FEATURES=SSE3,SSSE3,SSE41,POPCNT,SSE42

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python -c "import numpy; numpy.show_runtime()"

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>
```
or this,
```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET NPY_DISABLE_CPU_FEATURES="SSE3,SSSE3,SSE41,POPCNT,SSE42"

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python -c "import numpy; numpy.show_runtime()"
<frozen importlib._bootstrap>:219: RuntimeWarning: During parsing environment variable 'NPY_DISABLE_CPU_FEATURES':
You cannot disable CPU features ("SSE3 SSE42"), since they are not part of the dispatched optimizations
(SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512CD AVX512_SKX AVX512_CLX AVX512_CNL AVX512_ICL).

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>
```

**Comment by mattip** [comment]
@seiko2plus any idea why NPY_DISABLE_CPU_FEATURES is failing?

> Also, could you explain those commands? or just give me some resources where I can learn about those.

OPENBLAS_CORETYPE sets the CPU model used by OpenBLAS. One conjecture is that the core model is not being detected properly (you have not told us what hardware you are using). Haswell is a minimal model.

OPENBLAS_NUM_THREADS sets the maximum number of threads OpenBLAS will use when delegating work.

NPY_DISABLE_CPU_FEATURES will tell NumPy not to use certain CPU features. As I wrote [above](https://github.com/numpy/numpy/issues/23074#issuecomment-1400809136), NumPy will try to use low-level parallel routines. Calling one of these when the CPU does not support it will result in the program crashing.

**Comment by GurpreetSarangal** [comment]
Probably this is what you've asked for
```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>systeminfo

Host Name:                 GURPREET
OS Name:                   Microsoft Windows 7 Home Basic
OS Version:                6.1.7601 Service Pack 1 Build 7601
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          ajay
Registered Organization:
Product ID:                00346-OEM-8992752-50213
Original Install Date:     1/5/2012, 12:17:22 PM
System Boot Time:          1/23/2023, 4:33:56 PM
System Manufacturer:       LENOVO
System Model:              20060
System Type:               X86-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: x64 Family 6 Model 37 Stepping 5 GenuineIntel ~1317 Mhz
BIOS Version:              LENOVO 29CN31WW(V2.08), 9/7/2010
Windows Directory:         C:\windows
System Directory:          C:\windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC+05:30) Chennai, Kolkata, Mumbai, New Delhi
Total Physical Memory:     3,063 MB
Available Physical Memory: 823 MB
Virtual Memory: Max Size:  6,989 MB
Virtual Memory: Available: 3,036 MB
Virtual Memory: In Use:    3,953 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    PROGRAMERS
Logon Server:              \\GURPREET
Hotfix(s):                 424 Hotfix(s) Installed.
                           [01]: KB2849697
                           [02]: KB2849696
                           [03]: KB2841134
                           [04]: KB2670838
                           [05]: KB2830477
                           [06]: KB2592687
                           [07]: KB2305420
                           [08]: KB2393802
                           [09]: KB2425227
                           [10]: KB2479943
                           [11]: KB2484033
                           [12]: KB2488113
                           [13]: KB2491683
                           [14]: KB2492386
                           [15]: KB2502285
                           [16]: KB2503665
                           [17]: KB2505438
                           [18]: KB2506212
                           [19]: KB2506928
                           [20]: KB2507618
                           [21]: KB2509553
                           [22]: KB2510531
                           [23]: KB2511250
                           [24]: KB2511455
                           [25]: KB2515325
                           [26]: KB2522422
                           [27]: KB2529073
                           [28]: KB2532531
                           [29]: KB2533552
                           [30]: KB2536275
                           [31]: KB2536276
                           [32]: KB2541014
                           [33]: KB2544893
                           [34]: KB2545698
                           [35]: KB2547666
                           [36]: KB2552343
                           [37]: KB2560656
                           [38]: KB2563227
                           [39]: KB2564958
                           [40]: KB2567680
                           [41]: KB2570947
                           [42]: KB2574819
                           [43]: KB2579686
                           [44]: KB2584146
                           [45]: KB2585542
                           [46]: KB2588516
                           [47]: KB2604115
                           [48]: KB2618451
                           [49]: KB2619339
                           [50]: KB2620704
                           [51]: KB2620712
                           [52]: KB2621440
                           [53]: KB2631813
                           [54]: KB2633171
                           [55]: KB2633952
                           [56]: KB2639308
                           [57]: KB2639417
                           [58]: KB2640148
                           [59]: KB2641653
                           [60]: KB2641690
                           [61]: KB2644615
                           [62]: KB2647516
                           [63]: KB2647518
                           [64]: KB2647753
                           [65]: KB2653956
                           [66]: KB2654428
                           [67]: KB2655992
                           [68]: KB2656356
                           [69]: KB2656373
                           [70]: KB2656411
                           [71]: KB2658846
                           [72]: KB2659262
                           [73]: KB2660075
                           [74]: KB2660465
                           [75]: KB2660649
                           [76]: KB2665364
                           [77]: KB2667402
                           [78]: KB2675157
                           [79]: KB2676562
                           [80]: KB2677070
                           [81]: KB2679255
                           [82]: KB2685811
                           [83]: KB2685813
                           [84]: KB2685939
                           [85]: KB2686831
                           [86]: KB2688338
                           [87]: KB2690533
                           [88]: KB2691442
                           [89]: KB2695962
                           [90]: KB2698365
                           [91]: KB2699779
                           [92]: KB2705219
                           [93]: KB2709162
                           [94]: KB2709630
                           [95]: KB2712808
                           [96]: KB2718523
                           [97]: KB2718704
                           [98]: KB2719857
                           [99]: KB2719985
                           [100]: KB2726535
                           [101]: KB2727528
                           [102]: KB2729094
                           [103]: KB2729452
                           [104]: KB2731771
                           [105]: KB2731847
                           [106]: KB2732059
                           [107]: KB2732487
                           [108]: KB2732500
                           [109]: KB2735855
                           [110]: KB2736233
                           [111]: KB2736422
                           [112]: KB2741355
                           [113]: KB2742599
                           [114]: KB2743555
                           [115]: KB2750841
                           [116]: KB2756921
                           [117]: KB2757638
                           [118]: KB2758857
                           [119]: KB2761217
                           [120]: KB2763523
                           [121]: KB2770660
                           [122]: KB2773072
                           [123]: KB2785220
                           [124]: KB2786081
                           [125]: KB2786400
                           [126]: KB2789645
                           [127]: KB2798162
                           [128]: KB2799926
                           [129]: KB2800095
                           [130]: KB2803821
                           [131]: KB2807986
                           [132]: KB2808679
                           [133]: KB2813347
                           [134]: KB2813430
                           [135]: KB2813956
                           [136]: KB2820331
                           [137]: KB2832414
                           [138]: KB2833946
                           [139]: KB2834140
                           [140]: KB2834886
                           [141]: KB2835364
                           [142]: KB2836502
                           [143]: KB2836942
                           [144]: KB2836943
                           [145]: KB2839894
                           [146]: KB2840149
                           [147]: KB2840631
                           [148]: KB2843630
                           [149]: KB2844286
                           [150]: KB2845187
                           [151]: KB2846960
                           [152]: KB2847077
                           [153]: KB2847311
                           [154]: KB2847927
                           [155]: KB2849470
                           [156]: KB2852386
                           [157]: KB2853952
                           [158]: KB2855844
                           [159]: KB2857650
                           [160]: KB2859537
                           [161]: KB2861191
                           [162]: KB2861698
                           [163]: KB2861855
                           [164]: KB2862152
                           [165]: KB2862330
                           [166]: KB2862335
                           [167]: KB2862966
                           [168]: KB2862973
                           [169]: KB2863240
                           [170]: KB2864058
                           [171]: KB2864202
                           [172]: KB2868038
                           [173]: KB2868116
                           [174]: KB2868623
                           [175]: KB2868626
                           [176]: KB2868725
                           [177]: KB2871997
                           [178]: KB2872339
                           [179]: KB2876284
                           [180]: KB2876331
                           [181]: KB2882822
                           [182]: KB2884256
                           [183]: KB2887069
                           [184]: KB2888049
                           [185]: KB2891804
                           [186]: KB2892074
                           [187]: KB2893294
                           [188]: KB2893519
                           [189]: KB2894844
                           [190]: KB2898785
                           [191]: KB2898857
                           [192]: KB2900986
                           [193]: KB2901112
                           [194]: KB2904266
                           [195]: KB2908783
                           [196]: KB2909210
                           [197]: KB2909921
                           [198]: KB2911501
                           [199]: KB2912390
                           [200]: KB2913152
                           [201]: KB2913431
                           [202]: KB2913602
                           [203]: KB2916036
                           [204]: KB2918077
                           [205]: KB2918614
                           [206]: KB2919469
                           [207]: KB2922229
                           [208]: KB2925418
                           [209]: KB2926765
                           [210]: KB2928562
                           [211]: KB2929437
                           [212]: KB2929733
                           [213]: KB2929755
                           [214]: KB2929961
                           [215]: KB2930275
                           [216]: KB2931356
                           [217]: KB2936068
                           [218]: KB2937610
                           [219]: KB2939576
                           [220]: KB2943357
                           [221]: KB2952664
                           [222]: KB2953522
                           [223]: KB2957189
                           [224]: KB2957503
                           [225]: KB2957509
                           [226]: KB2957689
                           [227]: KB2961072
                           [228]: KB2961851
                           [229]: KB2962872
                           [230]: KB2966583
                           [231]: KB2968294
                           [232]: KB2970228
                           [233]: KB2971850
                           [234]: KB2972100
                           [235]: KB2972211
                           [236]: KB2972280
                           [237]: KB2973112
                           [238]: KB2973201
                           [239]: KB2973337
                           [240]: KB2973351
                           [241]: KB2976627
                           [242]: KB2976897
                           [243]: KB2977292
                           [244]: KB2977629
                           [245]: KB2977728
                           [246
Network Card(s):           4 NIC(s) Installed.
                           [01]: Realtek PCIe FE Family Controller
                                 Connection Name: Local Area Connection
                                 Status:          Media disconnected
                           [02]: Microsoft Virtual WiFi Miniport Adapter
                                 Connection Name: Wireless Network Connection 2
                                 Status:          Media disconnected
                           [03]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
                           [04]: Atheros AR9285 Wireless Network Adapter
                                 Connection Name: Wireless Network Connection
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.11.69
                                 IP address(es)
                                 [01]: 192.168.11.211
                                 [02]: fe80::5584:2ae0:696f:f8bb
                                 [03]: 2409:4055:615:e256:f995:95d2:fb71:53e1
                                 [04]: 2409:4055:615:e256:5584:2ae0:696f:f8bb

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>
```
well I think it went in too detail 😅

**Comment by mattip** [comment]
This is the relevant bit `x64 Family 6 Model 37 Stepping 5 GenuineIntel`. Looking [at WikiChip](https://en.wikichip.org/wiki/intel/cpuid#Family_6), Model 37 would appear to be a Westmere chip from ~2010. It precedes Haswell, which is the minimum chip we support. If you cannot find more modern hardware, I would suggest you use an online cloud platform like [GoogleColab](https://colab.research.google.com/) or [gradient](https://www.paperspace.com/gradient).

**Comment by seiko2plus** [comment]
> 
```(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET NPY_DISABLE_CPU_FEATURES="SSE3,SSSE3,SSE41,POPCNT,SSE42

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python -c "import numpy; numpy.show_runtime()"
<frozen importlib._bootstrap>:219: RuntimeWarning: During parsing environment variable 'NPY_DISABLE_CPU_FEATURES':
You cannot disable CPU features ("SSE3), since they are not part of the dispatched optimizations
(SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512CD AVX512_SKX AVX512_CLX AVX512_CNL AVX512_ICL).

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>
```

There is double quote added before the feature.
```
NPY_DISABLE_CPU_FEATURES="SSE3,SSSE3,SSE41,POPCNT,SSE42
```

**Comment by seiko2plus** [comment]
and here:
```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET NPY_DISABLE_CPU_FEATURES="SSE3,SSSE3,SSE41,POPCNT,SSE42"

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python -c "import numpy; numpy.show_runtime()"
<frozen importlib._bootstrap>:219: RuntimeWarning: During parsing environment variable 'NPY_DISABLE_CPU_FEATURES':
You cannot disable CPU features ("SSE3 SSE42"), since they are not part of the dispatched optimizations
(SSE3 SSSE3 SSE41 POPCNT SSE42 AVX F16C FMA3 AVX2 AVX512F AVX512CD AVX512_SKX AVX512_CLX AVX512_CNL AVX512_ICL).

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>
```
one double quote at the start and another at the end
```
NPY_DISABLE_CPU_FEATURES="SSE3,SSSE3,SSE41,POPCNT,SSE42"
```

**Comment by seiko2plus** [comment]
your emulator doesn't escape the quotes please avoid using them

**Comment by GurpreetSarangal** [comment]
I see. NumPY supports up to 4th gen processors of intel and I am using 3rd gen i3.  
Then how come the changes I made, resolved the crashing? and can I continue using numpy with my changes?  
I am asking this because currently, I am not in a position to buy a new laptop and I am also working on a computer vision project, for which I may also need to work offline. So, I could not always be online. I have also practiced a little bit with OpenCV and so far it is somehow working fine.

**Comment by GurpreetSarangal** [comment]
exactly @seiko2plus, actally that's why I tried these commands [here](#issuecomment-1401717528)

**Comment by seiko2plus** [comment]
@GurpreetSarangal,
What I meant is removes any quotes and it should works:
```
SET NPY_DISABLE_CPU_FEATURES=SSE3,SSSE3,SSE41,POPCNT,SSE42
```

**Comment by GurpreetSarangal** [comment]
@seiko2plus, this is the output with original code without any modifications   
```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET OPENBLAS_CORETYPE=Haswell

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET OPENBLAS_NUM_THREADS=1

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>SET NPY_DISABLE_CPU_FEATURES=SSE3,SSSE3,SSE41,POPCNT,SSE42

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python -c "import numpy; numpy.show_runtime()"

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>
```
   
and this is with my modifications

```
(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>python -c "import numpy; numpy.show_runtime()"
[{'simd_extensions': {'baseline': ['SSE', 'SSE2'],
                      'found': [],
                      'not_found': ['SSE3',
                                    'SSSE3',
                                    'SSE41',
                                    'POPCNT',
                                    'SSE42',
                                    'AVX',
                                    'F16C',
                                    'FMA3',
                                    'AVX2',
                                    'AVX512F',
                                    'AVX512CD',
                                    'AVX512_SKX',
                                    'AVX512_CLX',
                                    'AVX512_CNL',
                                    'AVX512_ICL']}},
 {'architecture': 'Nehalem',
  'filepath': 'C:\\Users\\ashok\\Documents\\_StudyMaterial\\Code\\Face '
              'Recognition - Open CV '
              'practice\\venv\\Lib\\site-packages\\numpy\\.libs\\libopenblas_v0.3.21-gcc_8_3_0.dll',
  'internal_api': 'openblas',
  'num_threads': 1,
  'prefix': 'libopenblas',
  'threading_layer': 'pthreads',
  'user_api': 'blas',
  'version': '0.3.21'}]

(venv) C:\Users\ashok\Documents\_StudyMaterial\Code\Face Recognition - Open CV practice>
```

**Comment by GurpreetSarangal** [comment]
@seiko2plus, the original code still crashes. It is probably because the reason what @mattip explained [here](#issuecomment-1404298302)

**Comment by GurpreetSarangal** [comment]
hey @mattip @seiko2plus, what do you think, can I use OpenCV with these modifications on my computer?? would it change anything??

**Comment by mattip** [comment]
Sorry @GurpreetSarangal I am not sure we can do much more to help you from afar. Can you search out a python user group in your area and they might be able to help you get going.

**Comment by mattip** [comment]
Closing as stale. Please reopen or open a new issue if you are still stuck.

**Comment by qubodup** [comment]
For posterity: the solution for me was to install `numpy==1.23.5` for Python 3.8 (which I use to support Windows 7).

```
pip install numpy==1.23.5
```

Starting with `1.24.0`, it will crash on an ASUS T-100 (64bit CPU, 32bit Windows 10 OS) with Atom CPU that does not support AVX instructions (Celeron and Pentium supposedly [also affected](https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#CPUs_with_AVX))

Interestingly in newer versions (3.11.4) using the latest available numpy (2.2.6) works.

I hope it's OK to document this little piece of info here, it is the 2nd top search result for "python import numpy crashes" but here we have Python 3.8 while https://github.com/numpy/numpy/issues/23864 involves 3.9, which I did not test.
